### ISSUE TITLE
it87 kernel module for fan control on a range of motherboards

### ISSUE THREAD

**Issue by trankillity** [ublue-os/akmods#191]
### Describe the package

I'd like to request the `it87` kernel module because `sensors` and `sensors-detect` doesn't expose my fans on my motherboard (Gigabyte B550i). I need this functionality because I have a deshrouded GPU and use mobo fan headers for my GPU fans. I can use [CoolerControl](https://gitlab.com/coolercontrol/coolercontrol) for that easily, however I need the sensors/control of those sensors via this module.

I was evaluating EndeavourOS and was able to install that from [AUR](https://aur.archlinux.org/packages/it87-dkms-git) by going down a [rabbit hole](https://wiki.archlinux.org/title/Lm_sensors#Gigabyte_B250/Z370/B450M/B560M/B660M/Z690_motherboards).

### Image

Bazzite

**Comment by austonpramodh** [comment]
Hi @trankillity I am looking for the same thing, I have A620I.

Did you manage to install it?

**Comment by trankillity** [comment]
No. The devs said it would need to be included as a module in the OS and requested that I put an issue up for it. No movement since then.

**Comment by austonpramodh** [comment]
I am not sure i understand 100%, devs asked you to create a new issue? or is it same issue.

I am not skilled with os stuff, but I could try a PR.

**Comment by trankillity** [comment]
They asked me to create this issue so that they could test whether including this module would cause any issues.

**Comment by austonpramodh** [comment]
Gotcha, I see. Cool thx

**Comment by StKob** [comment]
It affects Asus X470 Prime-PRO as well. There is a driver for Asus but it allows only readout of values through WMI.

**Comment by randogoth** [comment]
Would `acpi_ec` module work for your laptop? It's what NBFC is using to access fans. In that case #222  would be the solution. I have a similar issue and hoping for a fix

**Comment by trankillity** [comment]
> Would `acpi_ec` module work for your laptop? It's what NBFC is using to access fans. In that case [#222](https://github.com/ublue-os/akmods/issues/222) would be the solution. I have a similar issue and hoping for a fix

@randogoth - The original issue includes all the details, including the motherboard and the reasons this is needed. If you'd read it, you'd know that what you're saying is irrelevant because it is neither a laptop, nor is it MSI.

**Comment by romanalo** [comment]
I have the same Gigabyte B550i Aorus Pro motherboard. In my case it works fine with a configuration file:
/etc/sensors.d/gigabyte-it87.conf.
See the output of that file:

$ cat /etc/sensors.d/gigabyte-it87.conf 
chip "it8622-*"
    label in0 "Vcore"
    label in1 "+3.3V"
        compute in1 @ * 1.65, @ / 1.65
    label in2 "+12.0V"
        compute in2 @ * 6, @ / 6
    label in3 "+5.0V"
        compute in3 @ * 2.5, @ / 2.5
    label in4 "VSOC"
    label in5 "VDDP"
    label in6 "DRAM"
    label in7 "3VSB"
    label in8 "VBAT"
    label fan1 "CPU Fan"
    label fan2 "SYS Fan 1"
    label fan3 "SYS Fan 2"
    label temp1 "System"
    label temp2 "Chipset"
    label temp3 "CPU"
    label temp4 "PCIe x16"
    label temp5 "VRM"
    label temp6 "VSoC"
    label intrusion0 "Intrusion"
chip "gigabyte_wmi-*"
    label temp1 "System 1"
    label temp2 "Chipset PCh"
    label temp3 "CPU Soc."
    label temp4 "PCIe x16"
    label temp5 "VRM"
    label temp6 "VRM SoC"

**Comment by trankillity** [comment]
I no longer have this motherboard, so it would be a question for others who are also experiencing this issue.

**Comment by zerodarkzone** [comment]
Hi,

Could this be reopened?

I have a Gygabyte motherboard [X870E-AORUS-PRO-ICE](https://www.gigabyte.com/Motherboard/X870E-AORUS-PRO-ICE-rev-11) and I cannot read the sensor data.
